src/engine/test/source/hlp/hlp_test.cpp

```diff
@@ -173,3 +173,249 @@ TEST(map_test, incomplete_map_test)
+    ASSERT_EQ("2", resultMicros["timestamp.day"]);
+    ASSERT_EQ("15", resultMicros["timestamp.hour"]);
+    ASSERT_EQ("4", resultMicros["timestamp.minutes"]);
+    ASSERT_EQ("5", resultMicros["timestamp.seconds"]);
+
+    auto resultNanos = parseOp(stampnanoTs);
+    ASSERT_EQ("1", resultNanos["timestamp.month"]);
+    ASSERT_EQ("2", resultNanos["timestamp.day"]);
+    ASSERT_EQ("15", resultNanos["timestamp.hour"]);
+    ASSERT_EQ("4", resultNanos["timestamp.minutes"]);
+    ASSERT_EQ("5", resultNanos["timestamp.seconds"]);
+}
+
+TEST(hlpTimestampTests, Unix)
+{
+    static const char *logQl  = "[<timestamp>]";
+    static const char *unixTs = "[Mon Jan 2 15:04:05 MST 2006]";
+
+    auto parseOp = getParserOp(logQl);
+    auto result  = parseOp(unixTs);
+    ASSERT_EQ("2006", result["timestamp.year"]);
+    ASSERT_EQ("1", result["timestamp.month"]);
+    ASSERT_EQ("2", result["timestamp.day"]);
+    ASSERT_EQ("15", result["timestamp.hour"]);
+    ASSERT_EQ("4", result["timestamp.minutes"]);
+    ASSERT_EQ("5", result["timestamp.seconds"]);
+}
+
+TEST(hlpTimestampTests, specific_format)
+{
+    static const char *logQl =
+        "[<timestamp/RubyDate>] - "
+        "[<_ansicTs/timestamp/ANSIC>] - "
+        "[<_unixTs/timestamp/UnixDate>] - "
+        "[<_stampTs/timestamp/Stamp>]";
+    static const char *event =
+        "[Mon Jan 02 15:04:05 -0700 2006] - "
+        "[Mon Jan 2 15:04:05 2006] - "
+        "[Mon Jan 2 15:04:05 MST 2006] - "
+        "[Jan 2 15:04:05]";
+
+    auto parseOp = getParserOp(logQl);
+    auto result  = parseOp(event);
+
+    fprintf(stderr, "\n%30s | %s\n", "Key", "Val");
+    fprintf(stderr, "-------------------------------|------------\n");
+    for (auto const &r : result)
+    {
+        fprintf(stderr, "%30s | %s\n", r.first.c_str(), r.second.c_str());
+    }
+
+    ASSERT_EQ("2006", result["timestamp.year"]);
+    ASSERT_EQ("1", result["timestamp.month"]);
+    ASSERT_EQ("2", result["timestamp.day"]);
+    ASSERT_EQ("15", result["timestamp.hour"]);
+    ASSERT_EQ("4", result["timestamp.minutes"]);
+    ASSERT_EQ("5", result["timestamp.seconds"]);
+    ASSERT_EQ("-0700", result["timestamp.timezone"]);
+
+    ASSERT_EQ("2006", result["_ansicTs.year"]);
+    ASSERT_EQ("1", result["_ansicTs.month"]);
+    ASSERT_EQ("2", result["_ansicTs.day"]);
+    ASSERT_EQ("15", result["_ansicTs.hour"]);
+    ASSERT_EQ("4", result["_ansicTs.minutes"]);
+    ASSERT_EQ("5", result["_ansicTs.seconds"]);
+
+    ASSERT_EQ("2006", result["_unixTs.year"]);
+    ASSERT_EQ("1", result["_unixTs.month"]);
+    ASSERT_EQ("2", result["_unixTs.day"]);
+    ASSERT_EQ("15", result["_unixTs.hour"]);
+    ASSERT_EQ("4", result["_unixTs.minutes"]);
+    ASSERT_EQ("5", result["_unixTs.seconds"]);
+
+    ASSERT_EQ("1", result["_stampTs.month"]);
+    ASSERT_EQ("2", result["_stampTs.day"]);
+    ASSERT_EQ("15", result["_stampTs.hour"]);
+    ASSERT_EQ("4", result["_stampTs.minutes"]);
+    ASSERT_EQ("5", result["_stampTs.seconds"]);
+}
```

Comment:
Code style:
Add an extra line at the end of the file

Reply:
done